pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -637,23 +594,17 @@ private void addNewRow(int docId, GenericRow row) {
         // Update numValues info
         indexContainer._valuesInfo.updateSVNumValues();
 
-        MutableForwardIndex forwardIndex = indexContainer._forwardIndex;
+        MutableIndex forwardIndex = indexContainer._mutableIndexes.get(StandardIndexes.forward());
         FieldSpec fieldSpec = indexContainer._fieldSpec;
 
         DataType dataType = fieldSpec.getDataType();
         value = indexContainer._valueAggregator.getInitialAggregatedValue(value);
         switch (dataType.getStoredType()) {
           case INT:
-            forwardIndex.setInt(docId, ((Number) value).intValue());
-            break;
           case LONG:
-            forwardIndex.setLong(docId, ((Number) value).longValue());
-            break;
           case FLOAT:
-            forwardIndex.setFloat(docId, ((Number) value).floatValue());
-            break;
           case DOUBLE:
-            forwardIndex.setDouble(docId, ((Number) value).doubleValue());
+            forwardIndex.add(value, -1, docId);
```

Comment:
Now forwardIndex is responsible to do the switch. The reason why I keep the switch is to fail in the default case in order to do not break semantics (although I don't know if code actually can fall in the default case here)